apps/src/lib/kits/maker/connectToMakerBoard.js

```diff
@@ -1,16 +1,34 @@
 /**
  * @file Serialport scanning and board controller creation for Maker Toolkit.
  */
+import {getStore} from '../../../redux';
 import {findPortWithViableDevice} from './portScanning';
 import CircuitPlaygroundBoard from './CircuitPlaygroundBoard';
+import {
+  startConnecting,
+  reportConnected,
+  reportConnectionError,
+} from './redux';
 
 /**
  * Establish a connection to a maker board and return a controller for it.
  * @returns {Promise.<CircuitPlaygroundBoard>}}
  */
 export default function connectToMakerBoard() {
+  const store = getStore();
+  const dispatch = store.dispatch.bind(store);
+  let board;
   return Promise.resolve()
+      .then(() => dispatch(startConnecting()))
       .then(findPortWithViableDevice)
-      .then(port => new CircuitPlaygroundBoard(port))
-      .then(board => board.connect().then(() => board));
+      .then(foundPort => new CircuitPlaygroundBoard(foundPort))
+      .then(newBoard => board = newBoard)
+      .then(() => board.connect())
+      .then(() => dispatch(reportConnected()))
+      .then(() => board)
```

Comment:
I waffled on this for a while.  Given that I'm experiencing other limitations with promises (like how to 'cancel' out of a promise chain on user input), and some of the weirdness around testing promises we ran into recently, I now believe it might be best to minimize the number of `then` clauses.  Thanks for calling me out on this.

![waffle](https://cloud.githubusercontent.com/assets/1615761/24010038/f20d2758-0a33-11e7-8eff-10bb4af8fe49.jpg)